### PR TITLE
Rename `extrinsicData` into `field`

### PR DESCRIPTION
**Pull request by TotoGaz**
- renaming `namespace extrinsicMeshData` into namespace `fields`
- renaming `getExtrinsicData` into `getField`
- renaming `hasExtrinsicData` into `hasField`
- renaming `registerExtrinsicData` into `registerField`
- renaming `EXTRINSIC_MESH_DATA_TRAIT` into `DECLARE_FIELD`
- renaming files `*ExtrinsicData.hpp` into `*Fields.hpp` (files, include guards, doxygen)
- renamed variable names referring to extrinsic data.
- updated documentation

Fixes https://github.com/GEOSX/GEOSX/issues/2130

### CODE REVIEW

File: src/docs/sphinx/developerGuide/KeyComponents/WorkingWithData.rst
Comment (francoishamon) [inline src/docs/sphinx/developerGuide/KeyComponents/WorkingWithData.rst:79]: Maybe, on the line below and in the remainder of this page, you can replace `TotalDisplacement` with `totalDisplacement` to reflect the changes that was made in the code recently.

**Review by francoishamon — APPROVED** [review]
Thanks for doing that @TotoGaz, this is really nice now. I went through the files and did not spot any obvious mistake, so we can just merge this PR soon if the integrated tests are passing.

**Comment by rrsettgast** [comment]
@TotoGaz It would have been appropriate to have been given a chance to review this before you merged it.

**Comment by TotoGaz** [comment]
- I've discussed the subject live during the infrastructure meeting and the subject was validated.
- I've opened the issue https://github.com/GEOSX/GEOSX/issues/2130 13 days ago, presenting the changes that occurred in this PR and asking for comments and review even before I start the modifications (I remind it's just changing names). During those 2 weeks I only received comments by @francoishamon who's deeply involved in the subject.
- Therefore I asked @francoishamon to eventually review that what was in the PR was consistent with the proposition.

**Pull request merged**
